ADD: Переработка системы выбора корабля v1.0

Mirag1993 opened the pull request:
## Описание / Что этот PR делает
Полная переработка системы выбора кораблей, которая делает процесс вступления в экипаж намного удобнее и понятнее.
**Основные возможности:**
-Двухэтапный выбор кораблей: сначала фракция, потом корабль с группировкой по 8 фракциям
-Переделана система заявок на профессии: Apply → Pending → Approved/Denied → Select
-Современный UI/UX: карточки, бейджи, фильтрация, сортировка

Closed #1575

## Причина создания ПР / Почему это хорошо для игры
❌ Все корабли в одном списке - путаница и сложность поиска.
❌ Нет группировки по фракциям - сложно найти нужный тип корабля.
❌ Отсутствие системы заявок на конкретные профессии.

**Решение:**
✅ Улучшенная навигация - логичная группировка по 8 фракциям.
✅ Система заявок - подача заявок на конкретные профессии с отслеживанием статуса.
✅ Фильтрация и сортировка - по тегам, алфавиту, количеству экипажа.

## Список изменений
```yml
🆑
add: Полная переработка системы выбора кораблей с группировкой по фракциям.
add: Система фильтрации и сортировки кораблей.
add: Система заявок на конкретные профессии с отслеживанием статуса.
add: Современные TGUI интерфейсы ShipSelect и Application.
/🆑
```

Mirag1993 committed:
Переработка системы выбора корабля v1.0

-Добавлен выбор фракций
-Добавлен фильтр по тегам
-Добавлена сортировка по количеству экипажа , алфавиту

Mirag1993 committed:
Мини фикс 1.1

фикс мерцающего курсора

MysticalFaceLesS committed:
change png locate

Mirag1993 committed:
Мини фикс v2

Фиксим рантаймы при открытии описания

Mirag1993 committed:
Merge branch 'cersor-ship-selection' of https://github.com/Mirag1993/Shiptest-voyko into cersor-ship-selection

Mirag1993 committed:
Фикс v3

Улучшено отображение информации о корабле

Mirag1993 committed:
Фикс v4

Добавлено отображение тем фракций в выборе кораблей + мелкие фиксы

Mirag1993 committed:
Фикс v5

Огромный фикс , читай ридми

Mirag1993 committed:
Финальный

Mirag1993 committed:
Фикс в8 или 7

Фиксим несовместимости 516 и 515 бьенд , пришлось даунгрейдить код

Mirag1993 committed:
Фикс 8

Снова чиним ТГУИ

Mirag1993 committed:
Фикс 9

Фиксим отношение фракций

Mirag1993 committed:
Фикс 10

Вкладка все корабли - Other
Исправлен тултип

KOCMODECAHTHUK commented on tgui/packages/tgui/interfaces/EmotePanel.js:
```diff
@@ -1,7 +1,6 @@
 import { useBackend, useLocalState } from '../backend';
 import { Window } from '../layouts';
 import { Button, Section, Input, Stack, Icon } from '../components';
-import { capitalize } from 'common/string';
```
Лол

Mirag1993 replied:
Че за х

Mirag1993 committed:
Фикс 11 версия v1.01

Обновление цветов для тегов фракций , небольшой реворк в вкладке вступления на активный корабль , фикс тултипа , очиста модуля от хвостов

KOCMODECAHTHUK commented on tgui/packages/tgui/interfaces/EmotePanel.js:
```diff
@@ -127,7 +168,7 @@ export const EmotePanelContent = (props, context) => {
                     {emote.sound ? <Icon name="volume-up" /> : ''}
                     {emote.hands ? <Icon name="hand-paper" /> : ''}
                     {emote.targettable ? <Icon name="crosshairs" /> : ''}
-                    {emote.name}
```
But why? Ладно теперь это не просто смешно, а очень смешно.
Ибо теперь он не просто удалил строчку, а начал её использовать во всем файле.

Mirag1993 replied:
По сути он просто сделал что эмоции теперь будут писатся с большой буквы
бтв , мне кажется можно оставить

KOCMODECAHTHUK replied:
По сути Нейронка была права в том что эту штуку удалила?
Эмоуты специально с маленькой буквы пишутся.

Mirag1993 committed:
Фикс 12

Добавлено поля поиска по названию кораблей

Mirag1993 committed:
Фикс 13

Круглые кнопки

Mirag1993 committed:
Фикс 14

Тултипы теперь одинаковые

Mirag1993 committed:
Фикс 15

Добавлены 2 тема в тгуи задники , пираты и элизиум

Mirag1993 committed:
Фикс 16

Смена темы интекам и индам

MysticalFaceLesS committed:
color update

KOCMODECAHTHUK committed:
add: Замена лого всего шиптеста на наше

MysticalFaceLesS committed:
Merge branch 'beta-dev' of https://github.com/CeladonSS13/Shiptest into pr/Mirag1993/2002

Mirag1993 committed:
Фикс 17

Фиксим двойной клик (5 часов слез и стенаний)

Mirag1993 committed:
Фикс 18

Чутка переделаем джабки

Mirag1993 committed:
Фикс 19

Фикс общего окна выбора фракции

Mirag1993 committed:
Фикс 20

Доделал заявку на вступление в корабль , надеюсь на текущий момент это последний фикс

KOCMODECAHTHUK commented:
Не заходит при ОТКРЫТОМ кораблике вообще

Mirag1993 committed:
Merge branch 'CeladonSS13:beta-dev' into cersor-ship-selection

Mirag1993 committed:
Фикс 21

Починена возможность зайти в открытый корабль

KOCMODECAHTHUK commented on .gitignore:
```diff
@@ -202,3 +202,4 @@ check_regex_output.txt
 
 # Autowiki
 /tools/autowiki/node_modules
+SHIP_ZONES_PRACTICAL_EXAMPLES.md
```
СУКА

KOCMODECAHTHUK committed:
remove: FUCK AI GEN

Signed-off-by: KOCMOHABT <25280520+KOCMODECAHTHUK@users.noreply.github.com>

Mirag1993 committed:
Merge branch 'CeladonSS13:beta-dev' into cersor-ship-selection

Mirag1993 committed:
Merge branch 'CeladonSS13:beta-dev' into cersor-ship-selection

Mirag1993 committed:
Фикс 22

Много что сделано , ждем багрепорты

Mirag1993 committed:
Merge branch 'CeladonSS13:beta-dev' into cersor-ship-selection

Mirag1993 committed:
Фикс 23

Чиним назад кнопку назад

KOCMODECAHTHUK committed:
spellcheck: Правки коментариев, и доработка

KOCMODECAHTHUK committed:
spellcheck: Микрофикс

MysticalFaceLesS committed:
Update README.md

Signed-off-by: Voiko | Mr_DarkBladeS <48252701+MysticalFaceLesS@users.noreply.github.com>
